{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a6af40f8-b03d-406e-bc1b-a23d00e9b605",
   "metadata": {},
   "source": [
    "## Project Environment Setup\n",
    "\n",
    "We have downloaded metagenome data from NCBI to use for this project.\n",
    "\n",
    "There's no need to do this yourselves, but take note of this approach for future examination of metagenomes from NCBI's Short Read Archive (SRA)."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "045608d5-db42-4ea9-8272-67a24ff90a14",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "## Software Environment Setup\n",
    "\n",
    "The software needed for this tutorial can be found in the conda env config file ./coverm_env.yml\n",
    "\n",
    "It is pre-installed on the jupyter hub.  To activate it in the terminal type:\n",
    "\n",
    "```\n",
    "conda activate /mnt/storage/envs/coverm/\n",
    "```\n",
    "\n",
    "If you haven't done already, select the 'biopy' kernel on the upper right-hand side of the screen."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce8557d7-8055-4937-9e3c-53979503315f",
   "metadata": {},
   "source": [
    "## Data Setup\n",
    "\n",
    "For this lesson, we are going to use the SAGs that we will be working all week, **AG-910**, to examine their abundance over time at BATS using metagenomic read recruitment.  \n",
    "\n",
    "The metagenomes we will use are a small subset of metagenomes reported in [Biller et al., 2018](https://www.nature.com/articles/sdata2018176). These metagenomes are available through NCBI project ID [PRJNA385855](https://www.ncbi.nlm.nih.gov/bioproject?term=PRJNA385855). I've downloaded a metadata sheet with all sra metagenomes from this bioproject to: /mnt/storage/data/metagenomes/PRJNA385855_sra_metadata.csv\n",
    "(if you do not know how to download metadata tables from NCBI, I will be hapy to demonstrate; please let me know)\n",
    "\n",
    "Let's check this table out, and I'll show you which metagenomes I selected."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e70110c3-e849-4b74-8a2f-9d000c654330",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>AvgSpotLen</th>\n",
       "      <th>Bases</th>\n",
       "      <th>BioProject</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>BioSampleModel</th>\n",
       "      <th>bottle_id</th>\n",
       "      <th>Bytes</th>\n",
       "      <th>Center Name</th>\n",
       "      <th>...</th>\n",
       "      <th>lat_lon</th>\n",
       "      <th>Library Name</th>\n",
       "      <th>LibraryLayout</th>\n",
       "      <th>LibrarySelection</th>\n",
       "      <th>LibrarySource</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Platform</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>Sample Name</th>\n",
       "      <th>SRA Study</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR6507277</td>\n",
       "      <td>WGS</td>\n",
       "      <td>300</td>\n",
       "      <td>16133582100</td>\n",
       "      <td>PRJNA385855</td>\n",
       "      <td>SAMN08390922</td>\n",
       "      <td>MIMS.me,MIGS/MIMS/MIMARKS.water</td>\n",
       "      <td>2140200308</td>\n",
       "      <td>6618578156</td>\n",
       "      <td>MIT</td>\n",
       "      <td>...</td>\n",
       "      <td>22.75 N 158 W</td>\n",
       "      <td>S0627</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2018-05-01T00:00:00Z</td>\n",
       "      <td>S0627</td>\n",
       "      <td>SRP109831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR6507278</td>\n",
       "      <td>WGS</td>\n",
       "      <td>300</td>\n",
       "      <td>15874959000</td>\n",
       "      <td>PRJNA385855</td>\n",
       "      <td>SAMN08390923</td>\n",
       "      <td>MIMS.me,MIGS/MIMS/MIMARKS.water</td>\n",
       "      <td>2160200304</td>\n",
       "      <td>6562862443</td>\n",
       "      <td>MIT</td>\n",
       "      <td>...</td>\n",
       "      <td>22.75 N 158 W</td>\n",
       "      <td>S0628</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2018-05-01T00:00:00Z</td>\n",
       "      <td>S0628</td>\n",
       "      <td>SRP109831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR6507279</td>\n",
       "      <td>WGS</td>\n",
       "      <td>300</td>\n",
       "      <td>15069825300</td>\n",
       "      <td>PRJNA385855</td>\n",
       "      <td>SAMN08390924</td>\n",
       "      <td>MIMS.me,MIGS/MIMS/MIMARKS.water</td>\n",
       "      <td>1024800503</td>\n",
       "      <td>6265839401</td>\n",
       "      <td>MIT</td>\n",
       "      <td>...</td>\n",
       "      <td>31.66 N 64.16 W</td>\n",
       "      <td>S0629</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2018-05-01T00:00:00Z</td>\n",
       "      <td>S0629</td>\n",
       "      <td>SRP109831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR6507280</td>\n",
       "      <td>WGS</td>\n",
       "      <td>300</td>\n",
       "      <td>25807308000</td>\n",
       "      <td>PRJNA385855</td>\n",
       "      <td>SAMN08390925</td>\n",
       "      <td>MIMS.me,MIGS/MIMS/MIMARKS.water</td>\n",
       "      <td>1025200510</td>\n",
       "      <td>10523504402</td>\n",
       "      <td>MIT</td>\n",
       "      <td>...</td>\n",
       "      <td>31.66 N 64.16 W</td>\n",
       "      <td>S0630</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2018-05-01T00:00:00Z</td>\n",
       "      <td>S0630</td>\n",
       "      <td>SRP109831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR5720219</td>\n",
       "      <td>WGS</td>\n",
       "      <td>300</td>\n",
       "      <td>6713331000</td>\n",
       "      <td>PRJNA385855</td>\n",
       "      <td>SAMN07137016</td>\n",
       "      <td>MIMS.me,MIGS/MIMS/MIMARKS.water</td>\n",
       "      <td>1640201117</td>\n",
       "      <td>2811014041</td>\n",
       "      <td>MIT</td>\n",
       "      <td>...</td>\n",
       "      <td>22.75 N 158 W</td>\n",
       "      <td>S0519</td>\n",
       "      <td>PAIRED</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>METAGENOMIC</td>\n",
       "      <td>marine metagenome</td>\n",
       "      <td>ILLUMINA</td>\n",
       "      <td>2018-05-01T00:00:00Z</td>\n",
       "      <td>S0519</td>\n",
       "      <td>SRP109831</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Run Assay Type  AvgSpotLen        Bases   BioProject     BioSample  \\\n",
       "0  SRR6507277        WGS         300  16133582100  PRJNA385855  SAMN08390922   \n",
       "1  SRR6507278        WGS         300  15874959000  PRJNA385855  SAMN08390923   \n",
       "2  SRR6507279        WGS         300  15069825300  PRJNA385855  SAMN08390924   \n",
       "3  SRR6507280        WGS         300  25807308000  PRJNA385855  SAMN08390925   \n",
       "4  SRR5720219        WGS         300   6713331000  PRJNA385855  SAMN07137016   \n",
       "\n",
       "                    BioSampleModel   bottle_id        Bytes Center Name  ...  \\\n",
       "0  MIMS.me,MIGS/MIMS/MIMARKS.water  2140200308   6618578156         MIT  ...   \n",
       "1  MIMS.me,MIGS/MIMS/MIMARKS.water  2160200304   6562862443         MIT  ...   \n",
       "2  MIMS.me,MIGS/MIMS/MIMARKS.water  1024800503   6265839401         MIT  ...   \n",
       "3  MIMS.me,MIGS/MIMS/MIMARKS.water  1025200510  10523504402         MIT  ...   \n",
       "4  MIMS.me,MIGS/MIMS/MIMARKS.water  1640201117   2811014041         MIT  ...   \n",
       "\n",
       "           lat_lon Library Name LibraryLayout LibrarySelection LibrarySource  \\\n",
       "0    22.75 N 158 W        S0627        PAIRED           RANDOM   METAGENOMIC   \n",
       "1    22.75 N 158 W        S0628        PAIRED           RANDOM   METAGENOMIC   \n",
       "2  31.66 N 64.16 W        S0629        PAIRED           RANDOM   METAGENOMIC   \n",
       "3  31.66 N 64.16 W        S0630        PAIRED           RANDOM   METAGENOMIC   \n",
       "4    22.75 N 158 W        S0519        PAIRED           RANDOM   METAGENOMIC   \n",
       "\n",
       "            Organism  Platform           ReleaseDate Sample Name  SRA Study  \n",
       "0  marine metagenome  ILLUMINA  2018-05-01T00:00:00Z       S0627  SRP109831  \n",
       "1  marine metagenome  ILLUMINA  2018-05-01T00:00:00Z       S0628  SRP109831  \n",
       "2  marine metagenome  ILLUMINA  2018-05-01T00:00:00Z       S0629  SRP109831  \n",
       "3  marine metagenome  ILLUMINA  2018-05-01T00:00:00Z       S0630  SRP109831  \n",
       "4  marine metagenome  ILLUMINA  2018-05-01T00:00:00Z       S0519  SRP109831  \n",
       "\n",
       "[5 rows x 36 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "df = pd.read_csv(\"~/storage/data/metagenomes/PRJNA385855_sra_metadata.csv\", sep = \",\")\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41dddc92-6a9d-4b58-b05b-68e1b2bd2fd1",
   "metadata": {},
   "source": [
    "Let's check all column headers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "59ae8a60-e07a-424c-844d-7c30f58f5968",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Run', 'Assay Type', 'AvgSpotLen', 'Bases', 'BioProject', 'BioSample',\n",
       "       'BioSampleModel', 'bottle_id', 'Bytes', 'Center Name',\n",
       "       'Collection_date', 'Consent', 'cruise_id', 'DATASTORE filetype',\n",
       "       'DATASTORE provider', 'DATASTORE region', 'Depth', 'env_biome',\n",
       "       'env_feature', 'env_material', 'Experiment', 'geo_loc_name_country',\n",
       "       'geo_loc_name_country_continent', 'geo_loc_name', 'Instrument',\n",
       "       'isolation-source', 'lat_lon', 'Library Name', 'LibraryLayout',\n",
       "       'LibrarySelection', 'LibrarySource', 'Organism', 'Platform',\n",
       "       'ReleaseDate', 'Sample Name', 'SRA Study'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ada9cdc-bc6c-41cd-a44e-8d3d007283d4",
   "metadata": {},
   "source": [
    "This datasheet isn't only useful for metagenome selection, but it also has useful information on it that we can use for downstream analyses.\n",
    "\n",
    "Let's get a handle on the metagenomes described within this dataframe:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c93cbf74-6fe1-4853-a484-af5b0e316408",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Assay Type</th>\n",
       "      <th>Collection_date</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>Depth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR6507277</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009-08-19</td>\n",
       "      <td>HOT214</td>\n",
       "      <td>5m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR6507278</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009-11-04</td>\n",
       "      <td>HOT216</td>\n",
       "      <td>100m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR6507279</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009-07-14</td>\n",
       "      <td>BATS248</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR6507280</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2009-11-07</td>\n",
       "      <td>BATS252</td>\n",
       "      <td>100m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR5720219</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-10-10</td>\n",
       "      <td>HOT164</td>\n",
       "      <td>5m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SRR5720221</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-08-15</td>\n",
       "      <td>HOT162</td>\n",
       "      <td>175m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SRR5720223</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-09-28</td>\n",
       "      <td>HOT163</td>\n",
       "      <td>115m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SRR5720224</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-09-28</td>\n",
       "      <td>HOT163</td>\n",
       "      <td>175m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SRR5720226</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-06-15</td>\n",
       "      <td>HOT160</td>\n",
       "      <td>175m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SRR5720228</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-08-15</td>\n",
       "      <td>HOT162</td>\n",
       "      <td>100m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SRR5720230</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2003-07-15</td>\n",
       "      <td>BATS178</td>\n",
       "      <td>120m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SRR5720232</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2003-02-21</td>\n",
       "      <td>BATS173</td>\n",
       "      <td>40m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SRR5720234</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-12-08</td>\n",
       "      <td>BATS195</td>\n",
       "      <td>180m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SRR5720236</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2003-03-22</td>\n",
       "      <td>BATS174</td>\n",
       "      <td>180m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRR5720238</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2003-03-22</td>\n",
       "      <td>BATS174</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SRR5720240</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2003-01-17</td>\n",
       "      <td>HOT144</td>\n",
       "      <td>125m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>SRR5720242</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-10-10</td>\n",
       "      <td>HOT164</td>\n",
       "      <td>175m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>SRR5720244</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-11-10</td>\n",
       "      <td>HOT165</td>\n",
       "      <td>100m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>SRR5720246</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-12-14</td>\n",
       "      <td>HOT166</td>\n",
       "      <td>5m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>SRR5720249</td>\n",
       "      <td>WGS</td>\n",
       "      <td>2004-04-21</td>\n",
       "      <td>BATS187</td>\n",
       "      <td>40m</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Run Assay Type Collection_date cruise_id Depth\n",
       "0   SRR6507277        WGS      2009-08-19    HOT214    5m\n",
       "1   SRR6507278        WGS      2009-11-04    HOT216  100m\n",
       "2   SRR6507279        WGS      2009-07-14   BATS248   10m\n",
       "3   SRR6507280        WGS      2009-11-07   BATS252  100m\n",
       "4   SRR5720219        WGS      2004-10-10    HOT164    5m\n",
       "5   SRR5720221        WGS      2004-08-15    HOT162  175m\n",
       "6   SRR5720223        WGS      2004-09-28    HOT163  115m\n",
       "7   SRR5720224        WGS      2004-09-28    HOT163  175m\n",
       "8   SRR5720226        WGS      2004-06-15    HOT160  175m\n",
       "9   SRR5720228        WGS      2004-08-15    HOT162  100m\n",
       "10  SRR5720230        WGS      2003-07-15   BATS178  120m\n",
       "11  SRR5720232        WGS      2003-02-21   BATS173   40m\n",
       "12  SRR5720234        WGS      2004-12-08   BATS195  180m\n",
       "13  SRR5720236        WGS      2003-03-22   BATS174  180m\n",
       "14  SRR5720238        WGS      2003-03-22   BATS174    1m\n",
       "15  SRR5720240        WGS      2003-01-17    HOT144  125m\n",
       "16  SRR5720242        WGS      2004-10-10    HOT164  175m\n",
       "17  SRR5720244        WGS      2004-11-10    HOT165  100m\n",
       "18  SRR5720246        WGS      2004-12-14    HOT166    5m\n",
       "19  SRR5720249        WGS      2004-04-21   BATS187   40m"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['Run','Assay Type','Collection_date','cruise_id','Depth']].head(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4629227-ff50-425f-b745-bb4698d26a7f",
   "metadata": {},
   "source": [
    "As you might have noticed the codes names 'HOT' and 'BATS' are noted in the cruise_id column. Those are two time series stations.  HOT is in the Pacific, near Hawaii, and BATS is in the Atlantic, near Bermuda.  \n",
    "\n",
    "Our SAG plate is from BATS248. For this small exercise, we examine the abundance of AG-910 SAGs in the surface at BATS within this metagenome collection, so let's find just those metagenomes collected at BATS from either 1m or 10m depth."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d836e635-35a5-4f17-9083-50792b61df8f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 21 metagenomes that we are interested in downloading\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Collection_date</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Depth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>SRR5720233</td>\n",
       "      <td>2003-02-21</td>\n",
       "      <td>BATS173</td>\n",
       "      <td>SAMN07137079</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRR5720238</td>\n",
       "      <td>2003-03-22</td>\n",
       "      <td>BATS174</td>\n",
       "      <td>SAMN07137082</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>SRR5720327</td>\n",
       "      <td>2003-04-22</td>\n",
       "      <td>BATS175</td>\n",
       "      <td>SAMN07137064</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>SRR5720283</td>\n",
       "      <td>2003-05-20</td>\n",
       "      <td>BATS176</td>\n",
       "      <td>SAMN07137103</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SRR5720235</td>\n",
       "      <td>2003-07-15</td>\n",
       "      <td>BATS178</td>\n",
       "      <td>SAMN07137085</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>SRR5720286</td>\n",
       "      <td>2003-08-12</td>\n",
       "      <td>BATS179</td>\n",
       "      <td>SAMN07137088</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SRR5720332</td>\n",
       "      <td>2003-10-07</td>\n",
       "      <td>BATS181</td>\n",
       "      <td>SAMN07137067</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>SRR5720276</td>\n",
       "      <td>2003-11-04</td>\n",
       "      <td>BATS182</td>\n",
       "      <td>SAMN07137106</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>SRR5720262</td>\n",
       "      <td>2003-12-02</td>\n",
       "      <td>BATS183</td>\n",
       "      <td>SAMN07137109</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>SRR5720338</td>\n",
       "      <td>2004-01-27</td>\n",
       "      <td>BATS184</td>\n",
       "      <td>SAMN07137070</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>SRR5720322</td>\n",
       "      <td>2004-02-24</td>\n",
       "      <td>BATS185</td>\n",
       "      <td>SAMN07137121</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SRR5720337</td>\n",
       "      <td>2004-03-23</td>\n",
       "      <td>BATS186</td>\n",
       "      <td>SAMN07137073</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRR5720256</td>\n",
       "      <td>2004-04-21</td>\n",
       "      <td>BATS187</td>\n",
       "      <td>SAMN07137091</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>SRR5720257</td>\n",
       "      <td>2004-05-18</td>\n",
       "      <td>BATS188</td>\n",
       "      <td>SAMN07137112</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>SRR5720260</td>\n",
       "      <td>2004-06-15</td>\n",
       "      <td>BATS189</td>\n",
       "      <td>SAMN07137115</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SRR5720321</td>\n",
       "      <td>2004-08-17</td>\n",
       "      <td>BATS191</td>\n",
       "      <td>SAMN07137118</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>SRR5720251</td>\n",
       "      <td>2004-09-14</td>\n",
       "      <td>BATS192</td>\n",
       "      <td>SAMN07137094</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SRR5720307</td>\n",
       "      <td>2004-10-13</td>\n",
       "      <td>BATS193</td>\n",
       "      <td>SAMN07137097</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>SRR5720278</td>\n",
       "      <td>2004-11-12</td>\n",
       "      <td>BATS194</td>\n",
       "      <td>SAMN07137100</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>SRR5720342</td>\n",
       "      <td>2004-12-08</td>\n",
       "      <td>BATS195</td>\n",
       "      <td>SAMN07137076</td>\n",
       "      <td>1m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR6507279</td>\n",
       "      <td>2009-07-14</td>\n",
       "      <td>BATS248</td>\n",
       "      <td>SAMN08390924</td>\n",
       "      <td>10m</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Run Collection_date cruise_id     BioSample Depth\n",
       "74   SRR5720233      2003-02-21   BATS173  SAMN07137079    1m\n",
       "14   SRR5720238      2003-03-22   BATS174  SAMN07137082    1m\n",
       "119  SRR5720327      2003-04-22   BATS175  SAMN07137064   10m\n",
       "99   SRR5720283      2003-05-20   BATS176  SAMN07137103    1m\n",
       "75   SRR5720235      2003-07-15   BATS178  SAMN07137085   10m\n",
       "38   SRR5720286      2003-08-12   BATS179  SAMN07137088   10m\n",
       "64   SRR5720332      2003-10-07   BATS181  SAMN07137067    1m\n",
       "96   SRR5720276      2003-11-04   BATS182  SAMN07137106    1m\n",
       "90   SRR5720262      2003-12-02   BATS183  SAMN07137109    1m\n",
       "124  SRR5720338      2004-01-27   BATS184  SAMN07137070    1m\n",
       "117  SRR5720322      2004-02-24   BATS185  SAMN07137121    1m\n",
       "66   SRR5720337      2004-03-23   BATS186  SAMN07137073    1m\n",
       "22   SRR5720256      2004-04-21   BATS187  SAMN07137091    1m\n",
       "87   SRR5720257      2004-05-18   BATS188  SAMN07137112   10m\n",
       "89   SRR5720260      2004-06-15   BATS189  SAMN07137115   10m\n",
       "57   SRR5720321      2004-08-17   BATS191  SAMN07137118    1m\n",
       "83   SRR5720251      2004-09-14   BATS192  SAMN07137094    1m\n",
       "49   SRR5720307      2004-10-13   BATS193  SAMN07137097    1m\n",
       "97   SRR5720278      2004-11-12   BATS194  SAMN07137100    1m\n",
       "127  SRR5720342      2004-12-08   BATS195  SAMN07137076    1m\n",
       "2    SRR6507279      2009-07-14   BATS248  SAMN08390924   10m"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# dataframe of metagenomes of interest\n",
    "mgoi = df[df['cruise_id'].str.contains('BATS') & df['Depth'].isin(['10m','1m'])][['Run','Collection_date','cruise_id','BioSample','Depth']].sort_values(by = 'Collection_date')\n",
    "\n",
    "# going to save this table to file\n",
    "mgoi.to_csv(\"./bats_metagenomes_of_interest.csv\", index=False)\n",
    "\n",
    "print(\"There are\", len(mgoi), \"metagenomes that we are interested in downloading\")\n",
    "mgoi"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30a07e46-3ccb-4d9a-852c-84f7e2a3524b",
   "metadata": {},
   "source": [
    "Just for ease of access, I'm going to print out another small text file that has just the runids of the metagenomes we are interested in downloading into the directory we will download the files into:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7926d803-0d04-4376-a3a9-6010e5c50de6",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/mnt/storage/data/metagenomes/subsampled_metagenomes/metagenomes_to_download.txt', 'w') as oh:\n",
    "    for run in mgoi['Run']:\n",
    "        print(run, file = oh)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f198f67-ed68-4fc8-be0c-aa01a2187124",
   "metadata": {},
   "source": [
    "I'm next going to download these metagenomes from ncbi using the fastq-dump function from the sra-tools package.  Some parameters I'll use, I'm going to skip technical reads, and download only 1000000 reads from each metagenome."
   ]
  },
  {
   "cell_type": "raw",
   "id": "e0e6add6-8043-4c79-9e0c-bba18c281822",
   "metadata": {},
   "source": [
    "To download these metagenomes I opened up a temrinal window and ran:\n",
    "\n",
    "```\n",
    "conda activate /mnt/storage/envs/coverm/\n",
    "\n",
    "cd /mnt/storage/data/metagenomes/subsampled_metagenomes/\n",
    "\n",
    "while read p; do\n",
    "  fastq-dump --split-files --skip-technical -N 0 -X 1000000 --gzip --readids \"$p\"\n",
    "done < metagenomes_to_download.txt\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "543649c9-452a-488d-8dc9-2e8ab7b86678",
   "metadata": {},
   "source": [
    "Let's check which metagenomes were downloaded, using the python package 'glob' which identifies files using wildcards, and returns them as a list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8cf5ddbb-4b09-4208-8235-0f63da98ff90",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720233_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720307_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720338_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720332_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720257_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720322_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR6507279_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720262_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720238_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720260_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720321_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720286_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720327_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720278_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720256_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720342_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720251_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720276_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720283_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720337_1.fastq.gz',\n",
       " '/mnt/storage/data/metagenomes/subsampled_metagenomes/SRR5720235_1.fastq.gz']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from glob import glob\n",
    "\n",
    "mgs = glob('/mnt/storage/data/metagenomes/subsampled_metagenomes/*_1.fastq.gz')\n",
    "\n",
    "mgs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d7f1829-19ad-42d7-9af9-3a03260d98bd",
   "metadata": {},
   "source": [
    "Now I'm going to use what's called a 'list comprehension' to extract the runids from this list of files:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ebc512eb-3ed0-45cb-b7b3-b600ada2106c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "runids = [i.split(\"/\")[-1].split(\"_\")[0] for i in mgs]\n",
    "len(runids)"
   ]
  },
  {
   "cell_type": "raw",
   "id": "52f23606-629f-4b1b-93ce-ff2efebe068e",
   "metadata": {},
   "source": [
    "Verify if all metagenomes are downloaded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "a3737d78-a4df-4062-91b5-e21f1e7270be",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>Collection_date</th>\n",
       "      <th>cruise_id</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>Depth</th>\n",
       "      <th>downloaded</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>SRR5720233</td>\n",
       "      <td>2003-02-21</td>\n",
       "      <td>BATS173</td>\n",
       "      <td>SAMN07137079</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SRR5720238</td>\n",
       "      <td>2003-03-22</td>\n",
       "      <td>BATS174</td>\n",
       "      <td>SAMN07137082</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>SRR5720327</td>\n",
       "      <td>2003-04-22</td>\n",
       "      <td>BATS175</td>\n",
       "      <td>SAMN07137064</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>SRR5720283</td>\n",
       "      <td>2003-05-20</td>\n",
       "      <td>BATS176</td>\n",
       "      <td>SAMN07137103</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>SRR5720235</td>\n",
       "      <td>2003-07-15</td>\n",
       "      <td>BATS178</td>\n",
       "      <td>SAMN07137085</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>SRR5720286</td>\n",
       "      <td>2003-08-12</td>\n",
       "      <td>BATS179</td>\n",
       "      <td>SAMN07137088</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SRR5720332</td>\n",
       "      <td>2003-10-07</td>\n",
       "      <td>BATS181</td>\n",
       "      <td>SAMN07137067</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>SRR5720276</td>\n",
       "      <td>2003-11-04</td>\n",
       "      <td>BATS182</td>\n",
       "      <td>SAMN07137106</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>SRR5720262</td>\n",
       "      <td>2003-12-02</td>\n",
       "      <td>BATS183</td>\n",
       "      <td>SAMN07137109</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>SRR5720338</td>\n",
       "      <td>2004-01-27</td>\n",
       "      <td>BATS184</td>\n",
       "      <td>SAMN07137070</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>SRR5720322</td>\n",
       "      <td>2004-02-24</td>\n",
       "      <td>BATS185</td>\n",
       "      <td>SAMN07137121</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SRR5720337</td>\n",
       "      <td>2004-03-23</td>\n",
       "      <td>BATS186</td>\n",
       "      <td>SAMN07137073</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>SRR5720256</td>\n",
       "      <td>2004-04-21</td>\n",
       "      <td>BATS187</td>\n",
       "      <td>SAMN07137091</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>SRR5720257</td>\n",
       "      <td>2004-05-18</td>\n",
       "      <td>BATS188</td>\n",
       "      <td>SAMN07137112</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>SRR5720260</td>\n",
       "      <td>2004-06-15</td>\n",
       "      <td>BATS189</td>\n",
       "      <td>SAMN07137115</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>SRR5720321</td>\n",
       "      <td>2004-08-17</td>\n",
       "      <td>BATS191</td>\n",
       "      <td>SAMN07137118</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>SRR5720251</td>\n",
       "      <td>2004-09-14</td>\n",
       "      <td>BATS192</td>\n",
       "      <td>SAMN07137094</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>SRR5720307</td>\n",
       "      <td>2004-10-13</td>\n",
       "      <td>BATS193</td>\n",
       "      <td>SAMN07137097</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>SRR5720278</td>\n",
       "      <td>2004-11-12</td>\n",
       "      <td>BATS194</td>\n",
       "      <td>SAMN07137100</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>SRR5720342</td>\n",
       "      <td>2004-12-08</td>\n",
       "      <td>BATS195</td>\n",
       "      <td>SAMN07137076</td>\n",
       "      <td>1m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR6507279</td>\n",
       "      <td>2009-07-14</td>\n",
       "      <td>BATS248</td>\n",
       "      <td>SAMN08390924</td>\n",
       "      <td>10m</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Run Collection_date cruise_id     BioSample Depth downloaded\n",
       "74   SRR5720233      2003-02-21   BATS173  SAMN07137079    1m        yes\n",
       "14   SRR5720238      2003-03-22   BATS174  SAMN07137082    1m        yes\n",
       "119  SRR5720327      2003-04-22   BATS175  SAMN07137064   10m        yes\n",
       "99   SRR5720283      2003-05-20   BATS176  SAMN07137103    1m        yes\n",
       "75   SRR5720235      2003-07-15   BATS178  SAMN07137085   10m        yes\n",
       "38   SRR5720286      2003-08-12   BATS179  SAMN07137088   10m        yes\n",
       "64   SRR5720332      2003-10-07   BATS181  SAMN07137067    1m        yes\n",
       "96   SRR5720276      2003-11-04   BATS182  SAMN07137106    1m        yes\n",
       "90   SRR5720262      2003-12-02   BATS183  SAMN07137109    1m        yes\n",
       "124  SRR5720338      2004-01-27   BATS184  SAMN07137070    1m        yes\n",
       "117  SRR5720322      2004-02-24   BATS185  SAMN07137121    1m        yes\n",
       "66   SRR5720337      2004-03-23   BATS186  SAMN07137073    1m        yes\n",
       "22   SRR5720256      2004-04-21   BATS187  SAMN07137091    1m        yes\n",
       "87   SRR5720257      2004-05-18   BATS188  SAMN07137112   10m        yes\n",
       "89   SRR5720260      2004-06-15   BATS189  SAMN07137115   10m        yes\n",
       "57   SRR5720321      2004-08-17   BATS191  SAMN07137118    1m        yes\n",
       "83   SRR5720251      2004-09-14   BATS192  SAMN07137094    1m        yes\n",
       "49   SRR5720307      2004-10-13   BATS193  SAMN07137097    1m        yes\n",
       "97   SRR5720278      2004-11-12   BATS194  SAMN07137100    1m        yes\n",
       "127  SRR5720342      2004-12-08   BATS195  SAMN07137076    1m        yes\n",
       "2    SRR6507279      2009-07-14   BATS248  SAMN08390924   10m        yes"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mgoi.loc[~mgoi['Run'].isin(runids),'downloaded'] = 'no'\n",
    "mgoi['downloaded'] = mgoi['downloaded'].fillna('yes')\n",
    "mgoi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "6cfde1fa-a680-4936-9770-0b3284bad498",
   "metadata": {},
   "outputs": [],
   "source": [
    "mgoi.to_csv(\"/mnt/storage/data/metagenomes/bats_metagenomes_of_interest.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5371c1c0-43db-4640-97c9-85d001e185e7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "biopy",
   "language": "python",
   "name": "biopy"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
